iothub/device/src/Transport/ProtocolRoutingDelegatingHandler.cs

```diff
@@ -68,6 +69,11 @@ public override async Task OpenAsync(CancellationToken cancellationToken)
                 {
                     CreateNewTransportIfNotReady();
                     await base.OpenAsync(cancellationToken).ConfigureAwait(false);
+                    if (_disposed)
```

Comment:
This isn't interlocked so it won't work: Dispose will change the flag on its thread but it won't change value here.
We could use an interlocked / volatile or a lock although I would like to avoid locks in Dispose.

Reply:
Please add a comment on why we're doing this (to allow safe Dispose).

Reply:
done